### PR TITLE
Start making releases

### LINKED ISSUE #180: start making releases
I was thinking, what about starting making tagged releases  (or even just tagging on github) ?

It's a bit of work, but it would :

- make the install process a bit easier
- simplify third-party packaging (I'm thinking about https://github.com/YunoHost-Apps/ihatemoney_ynh )
- force us to keep track of changes in a ChangeLog

What do you think ?

### PR DESCRIPTION
**Pull request by almet**
This documents how we could start to do releases. I've yet to try the process, but this is open for feedback! (cc @JocelynDelalande).

Closes #180

### CODE REVIEW

File: CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Adam Chainz <adam@adamj.eu>
+Adrien CLERC <adrien@antipoul.fr>
+Alexis Metaireau <alexis@notmyidea.org>
+Arnaud Bos <arnaud.tlse@gmail.com>
+Baptiste Jonglez <git@bitsofnetworks.org>
+Berteh <berteh@gmail.com>
+Feth AREZKI <feth@tuttu.info>
+fredericsureau <frederic.sureau@gmail.com>
+Frédéric Sureau <fred@milka.(none)>
+JocelynDelalande <JocelynDelalande@users.noreply.github.com>
```
Comment (JocelynDelalande) [inline CONTRIBUTORS.rst]: You could remove this one, that's me as well

File: CONTRIBUTORS.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Jocelyn Delande <jocelyn@crapouillou.net>
```
Comment (JocelynDelalande) [inline CONTRIBUTORS.rst]: Typo on my name (I know, that's my fault in first place)

File: README.rst [hunk 3]
```diff
@@ -98,3 +98,40 @@ End-user
...
+
+    $ pip install "zest.releaser"
+
+- Merge remaining pull requests
+- Update ``CHANGELOG.rst``
+- Update :file:`CONTRIBUTORS.rst`. The following hairy command will output the full list:
+
+.. code-block:: bash
+
+     $ git shortlog -sne | awk '{$1=""; sub(" ", ""); print}' | awk -F'<' '!x[$1]++' | awk -F'<' '!x[$2]++' | sort
```
Comment (JocelynDelalande) [inline README.rst]: This one liner does not find my actual name *« Jocelyn Delalande <jocelyn@crapouillou.net> »*

**Comment by almet** [comment]
Okay, I did some packaging for this (setup.py and manifest.in). It's probably still rough to the edges, but seems to work.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+
+setup(name='ihatemoney',
+      version='1.0.0',
+      description='A simple shared budget manager web application.',
+      long_description="{}\n\n{}".format(README, CHANGELOG),
+      license='Custom BSD Beerware',
+      classifiers=[
+          "Programming Language :: Python",
```
Comment (JocelynDelalande) [inline setup.py]: Python 2.7 is not mentioned ?

Comment (almet) [inline setup.py]: Updated :)

**Comment by almet** [comment]
@JocelynDelalande are you okay with the current proposal?

**Comment by JocelynDelalande** [comment]
Thanks for the edits :)

> @JocelynDelalande are you okay with the current proposal?

I'm unconfortable with the situation where we publish to pypi something non functional (see #187).

So I would prefer that we merge that PR *but* without publishing to pypi for the moment. One step at a time :-)

**Comment by almet** [comment]
Okay, let's merge this as-is then and iterate! Thanks for the feedback.

**Comment by JocelynDelalande** [comment]
@almet cool :-) still we have that [non-functional and undocumented pypi release](https://pypi.python.org/pypi/ihatemoney/1.0.0).

**Comment by almet** [comment]
I just removed it, thanks!

**Pull request merged**